lib/spack/spack/package_base.py

```diff
@@ -646,6 +647,9 @@ class PackageBase(WindowsRPath, PackageViewMixin, metaclass=PackageMeta):
     #: TestSuite instance used to manage stand-alone tests for 1+ specs.
     test_suite: Optional["TestSuite"] = None
 
+    depends_on("gcc-runtime", type="link", when="%gcc platform=linux")
+    depends_on("gcc-runtime", type="link", when="%gcc platform=cray")
```

Comment:
Is there a way to add %gcc@current_version?

Reply:
I am not sure the compiler property of PackageBase is usable util the class is instantiated.

Reply:
How would you want to use it?

Reply:
depends_on("gcc-runtime%current_compiler_and_version", type="link", when="%gcc platform=linux")

Reply:
We usually use the Spack built compiler for cxxstd=20 builds.

Reply:
Right, yeah, I think the next step would be to say `provides("libstdcxx@...")` and then make the depends on slightly more generic, but internally you could use what you're saying yes.